languageFactories/language-factory-core/src/main/scala/cromwell/languages/util/ResolvedImportsStore.scala

```diff
@@ -0,0 +1,20 @@
+package cromwell.languages.util
+
+
+class ResolvedImportsStore {
+  private val mutex = new Object
+  private var resolvedImportsSet = Set.empty[String]
+
+  def updateResolvedImportsSet(importPath: String): Unit = {
```

Comment:
Is there any way we can make this an immutable data structure and replace this method with something that returns a new `ResolvedImportsStore` with the combined contents of the existing imports and the new path?

I'm not against mutexes but I favor lazier, simpler to understand approaches when possible